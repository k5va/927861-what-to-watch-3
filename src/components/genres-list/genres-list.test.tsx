import * as renderer from "react-test-renderer";
import GenresList from "./genres-list";

const genres = [`Action`, `Comedy`, `Drama`, `Thriller`];

const selectedGenre = `Drama`;

const onGenreSelect = jest.fn();

it(`GenresList should render correctly`, () => {
  const wrapper = renderer
    .create(
        <GenresList genres={genres} selectedGenre={selectedGenre} onGenreSelect={onGenreSelect} />,
        {
          createNodeMock: () => ({})
        }
    )
    .toJSON();
  expect(wrapper).toMatchSnapshot();
});
