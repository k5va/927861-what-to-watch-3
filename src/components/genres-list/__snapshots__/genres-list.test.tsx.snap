// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenresList should render correctly 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item "
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      All genres
    </a>
  </li>
  <li
    className="catalog__genres-item "
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Action
    </a>
  </li>
  <li
    className="catalog__genres-item "
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Comedy
    </a>
  </li>
  <li
    className="catalog__genres-item catalog__genres-item--active"
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Drama
    </a>
  </li>
  <li
    className="catalog__genres-item "
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Thriller
    </a>
  </li>
</ul>
`;
